src/AlertsManagement/AlertsManagement/AlertCommands/MeasureAzureAlertStatistic.cs

```diff
@@ -0,0 +1,149 @@
+        public string GroupBy { get; set; }
+
+        /// <summary>
+        /// Resource Id
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on Resource Id of the target resource of alert.")]
+        public string TargetResource { get; set; }
+
+        /// <summary>
+        /// Resource Type
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on Resource type of the target resource of alert.")]
+        [ResourceTypeCompleter]
+        public string TargetResourceType { get; set; }
+
+        /// <summary>
+        /// Resource Group Name
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on Resource group name of the target resource of alert.")]
+        [ResourceGroupCompleter]
+        public string TargetResourceGroup { get; set; }
+
+        /// <summary>
+        /// Monitor Service
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on Moniter Service")]
+        [PSArgumentCompleter("Platform", "Log Analytics", "SCOM", "Activity Log")]
+        public string MonitorService { get; set; }
+
+        /// <summary>
+        /// Monitor Condition
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on Monitor Condition")]
+        [PSArgumentCompleter("Fired", "Resolved")]
+        public string MonitorCondition { get; set; }
+
+        /// <summary>
+        /// Severity
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on Severity of alert")]
+        [PSArgumentCompleter("Sev0", "Sev1", "Sev2", "Sev3", "Sev4")]
+        public string Severity { get; set; }
+
+        /// <summary>
+        /// Alert State
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on State of alert")]
+        [PSArgumentCompleter("New", "Acknowledged", "Closed")]
+        public string State { get; set; }
+
+        /// <summary>
+        /// Alert Rule Id
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on Alert Rule Id")]
+        public string AlertRuleId { get; set; }
+
+        /// <summary>
+        /// Time range
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Supported time range values – 1h, 1d, 7d, 30d (Default is 1d)")]
+        [PSArgumentCompleter("1h", "1d", "7d", "30d")]
+        public string TimeRange { get; set; }
+
+        /// <summary>
+        /// Custom time range
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Supported format - <start-time>/<end-time> where time is in ISO-8601 format")]
+        public string CustomTimeRange { get; set; }
```

Comment:
This should be a TimeSpan, and your swagger should be marked to show this as a duration, which will automatically be represented by the appropriate type in each language (TiemSpan in .Net) and translated to ISO8601 format

Reply:
We are not planning for any change in swagger as of now to avoid any breaking change. What I could do is take Timespan as input in powershell and convert it as required by the API? That should be fine, right?

Reply:
No, this is actually a problem with your API cross-platform, as it will be difficult for SDKs in every language to generate correctly.  This is important to fix.  For the purposes of this preview, you cna use the timespan in powershell and convert it, but you must change the swagger spec.

Note that this is adding a format to the string in Swagger, it is not changing a type - this engages the parituclar mechanism for dealing with ISO8601 strings in each SDK.